test/e2e/framework/network_pods.go

```diff
@@ -86,3 +88,48 @@ func (f *Framework) CreateTCPCheckConnectorPod(cluster int, remoteCheckPod *v1.P
 	Expect(err).NotTo(HaveOccurred())
 	return tcpCheckPod
 }
+
+type TestPodScheduling int
+
+const (
+	InvalidScheduling TestPodScheduling = iota
+	GatewayNode
+	NonGatewayNode
+)
+
+func nodeAffinity(scheduling TestPodScheduling) *v1.Affinity {
+
+	const gatewayLabel = "submariner.io/gateway"
+
+	var nodeSelReqs []v1.NodeSelectorRequirement
+
+	switch scheduling {
+	case GatewayNode:
+		addNodeSelectorRequirement(&nodeSelReqs, gatewayLabel, v1.NodeSelectorOpIn, []string{"true"})
+
+	case NonGatewayNode:
+		addNodeSelectorRequirement(&nodeSelReqs, gatewayLabel, v1.NodeSelectorOpDoesNotExist, nil)
+		addNodeSelectorRequirement(&nodeSelReqs, gatewayLabel, v1.NodeSelectorOpNotIn, []string{"true"})
+	}
+
+	affinity := v1.Affinity{
+		NodeAffinity: &v1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: nodeSelReqs,
+					},
+				},
+			},
+		},
+	}
+	return &affinity
+}
+
+func addNodeSelectorRequirement(nodeSelReqs *[]v1.NodeSelectorRequirement, label string, op v1.NodeSelectorOperator, values []string) {
+	*nodeSelReqs = append(*nodeSelReqs, v1.NodeSelectorRequirement{
+		Key:      label,
+		Operator: v1.NodeSelectorOpNotIn,
+		Values:   []string{"true"},
+	})
```

Comment:
Minor thing but I think it's more intuitive and idiomatic in Go to return the new `[]v1.NodeSelectorRequirement` from the function rather than via a formal parameter.

Reply:
I'll change it, I was trying to avoid code repetition as much as possible. I guess that pattern is good for parallelization.